astropy/coordinates/representation.py

```diff
@@ -171,6 +171,19 @@ def __init__(self, *args, **kwargs):
         for component, attr in zip(components, attrs):
             setattr(self, '_' + component, attr)
 
+    def __eq__(self, other):
+        if not isinstance(other, BaseRepresentationOrDifferential):
+            return False
```

Comment:
This is wrong: should return `NotImplemented` (CPython will convert to `False`).